src/main/java/com/github/firmwehr/gentle/GentleCompiler.java

```diff
@@ -136,16 +101,16 @@ private static void parseTestCommand(Path path) {
 			Parser parser = Parser.fromLexer(source, lexer);
 			parser.parse(); // result ignored for now
 		} catch (IOException e) {
-			LOGGER.error("Could not read file '{}': {}", path, e.getMessage());
+			UserOutput.getInstance().userError("Could not read file '%s': %s", path, e.getMessage());
 			System.exit(1);
 		} catch (SourceException e) {
-			LOGGER.error("Error reading file '{}': {}", path, e.getMessage());
+			UserOutput.getInstance().userError("Error reading file '%s': %s", path, e.getMessage());
 			System.exit(1);
 		} catch (LexerException e) {
-			LOGGER.error("Lexing failed", e);
+			UserOutput.getInstance().userError(e.getMessage());
 			System.exit(1);
 		} catch (ParseException e) {
-			LOGGER.error("Parsing failed", e);
+			UserOutput.getInstance().userError(e.getMessage());
```

Comment:
Given how often this pattern is used, what about an `userError(Throwable)` overload?

Reply:
It only prints the message of the throwable so it might be a bit more confusing to have an overload for it?

Reply:
How about `userErrorFromException` to indicate that the output is only derived from the given exception?